### DISCUSSION TITLE
How Can i send a request to update status.

### DISCUSSION THREAD

**Discussion by bestony** [inertiajs/inertia#546] [Help (Vue)]
Hi, I want to send request to update use state (like heartbeat request)

What should i do in Inertia.js  & Laravel ?

## Why Not Normal Inertia request?

Inertia Request must return a Inertia Response, if not ,it will show return msg on screen, i can't just return a "ok" in my function.

## I tried this methods
 In Controller
```php
public function index(){
   // logic
     return Inertia::render("pages",[
       "data" => "x" 
      ]);
 } 
public function updateState(){
   // logic
    return Inertia::render("pages",[
     "msg => "ok" 
    ]);
}
```
In Page Component

```js
this.$inertia.patch(
        route("room.updateRoomUserProps", this.room.slug),
        {
          id,
        },
        {
          replace: true,
          preserveState: true,
          preserveScroll: true,
          only: ["msg"],
        }
      );
```

## my question

it works, but i think it's ugly, is here any better way to do it ?

**Comment by lostdesign** [comment]
Why not make a normal API Endpoint that has nothing to do with Inertia? Something like a simple endpoint that returns the status which you call by using axios, fetch or any other HTTP Call? With response json you could return plain json. That should do the trick no?

**Reply by bestony** [reply]
yes, i can do it. may be inertiajs don't have a methods for this situation.
